src/main/java/xyz/nucleoid/plasmid/mixin/server/onJoinMixin.java

```diff
@@ -0,0 +1,22 @@
+package xyz.nucleoid.plasmid.mixin.server;
+
+import net.minecraft.network.ClientConnection;
+import net.minecraft.server.PlayerManager;
+import net.minecraft.server.network.ServerPlayerEntity;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import xyz.nucleoid.plasmid.storage.FriendSystem.FriendList;
+import xyz.nucleoid.plasmid.storage.FriendSystem.FriendListManager;
+
+@Mixin(PlayerManager.class)
+public class onJoinMixin<class_3222> {
+    @Inject(method = "onPlayerConnect", at = @At("RETURN"))
+    private void injectMethod(ClientConnection connection, ServerPlayerEntity player, CallbackInfo info) {
+        System.out.println("Player connected");
+        FriendList temp = new FriendList();
+        FriendListManager.appendNewFriendList(player.getUuid(), temp);
```

Comment:
It might be worth handling this creation logic on-demand, so as to only create the friend list when it is required.

Reply:
i did it this way, because maybe in the future (if we get have a DB to save Friends this can be used to actually load data onto the list)